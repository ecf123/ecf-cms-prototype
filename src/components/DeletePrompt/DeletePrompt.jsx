import "./DeletePrompt.scss";
import warning from "../../assets/exclamation-mark-icon.svg";

const DeletePrompt = ({ title, handleCancel, handleDelete }) => {
  {/*This is the component for the delete prompt box, it will need props for the handling of each click
    as well as a prop for the title of what is being deleted.*/}
  return (
    <div className="delete-prompt">
      <div className="delete-prompt__image-conatiner">
        <img src={warning} alt="Warning Image" className="delete-prompt__image" />
      </div>
      <h2 className="delete-prompt__title">Delete {title}</h2>
      <p className="delete-prompt__content">Are you sure you want to delete this {title}? This action cannot be undone.</p>
      <div className="delete-prompt__button-holder">
        <button onClick={handleCancel}>Cancel</button>
        <button onClick={handleDelete}>Delete</button>
      </div>
    </div>
  );
};

export default DeletePrompt;
