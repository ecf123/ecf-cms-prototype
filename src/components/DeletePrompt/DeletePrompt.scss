@use "../../styles/main.scss" as *;
@use "../../styles/variables.scss" as *;

.delete-prompt {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: fixed;
  margin: auto;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 5;
  width: 25rem;
  height: 16.75rem;
  border-radius: 0.75rem;
  background: $color-white;
  padding: 1.5rem;
  &__image-conatiner {
    display: grid;
    height: 2.5rem;
    width: 2.5rem;
    background: #fef3f2;
    border-radius: 1.75rem;
  }
  &__image {
    margin: 14.8%;
    height: 1.75rem;
    width: 1.75rem;
    border-radius: 1.75rem;
    padding: 0.5rem;
    background-color: #fee4e2;
  }
  &__title {
    font-size: $font-size-medium;
    color: $color-black;
    font-family: $font-secondary;
    letter-spacing: $letter-spacing;
    font-weight: $font-weight-regular;
  }
  &__content {
    font-size: $font-size-extra-small;
    color: $color-grey-dark;
    font-family: $font-secondary;
    letter-spacing: $letter-spacing;
  }
  &__button-holder {
    display: flex;
    justify-content: space-evenly;
  }
}

@media screen and (max-width: $mobile-breakpoint) {
  .delete-prompt {
    display: flex;
    flex-direction: column;
    position: fixed;
    margin: auto;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 5;
    width: 20rem;
    height: 16.75rem;
    border-radius: 0.75rem;
    background: $color-white;
    padding: 1.5rem;
    gap: 2rem;
    font-family: $font-secondary;
    letter-spacing: $letter-spacing;

    &__image-conatiner {
      display: grid;
      height: 2.5rem;
      width: 2.5rem;
      background: #fef3f2;
      border-radius: 1.75rem;
    }
    &__image {
      margin: 14.8%;
      height: 1.75rem;
      width: 1.75rem;
      border-radius: 1.75rem;
      padding: 0.5rem;
      background-color: #fee4e2;
    }
    &__title {
      font-size: $font-size-medium;
      color: $color-black;
    }
    &__content {
      font-size: $font-size-extra-small;
      color: $color-grey-dark;
    }
    &__button-holder {
      display: flex;
      justify-content: space-evenly;
    }
  }
}
