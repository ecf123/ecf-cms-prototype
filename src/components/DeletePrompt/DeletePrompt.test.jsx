import { customRender } from "../../utils/testUtils";
import DeletePrompt from "./DeletePrompt.jsx";

it("Should render DeletePromt with children", () => {
  const { container } = customRender(
    <DeletePrompt>
      <p>some content</p>
    </DeletePrompt>
  );

  expect(container).toMatchSnapshot();
});

