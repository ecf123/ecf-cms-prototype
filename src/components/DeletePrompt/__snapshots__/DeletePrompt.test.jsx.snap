// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DeletePrompt with children 1`] = `
<div>
  <div
    class="delete-prompt"
  >
    <div
      class="delete-prompt__image-conatiner"
    >
      <img
        alt="Warning Image"
        class="delete-prompt__image"
        src="exclamation-mark-icon.svg"
      />
    </div>
    <h2
      class="delete-prompt__title"
    >
      Delete 
      Title
    </h2>
    <p
      class="delete-prompt__content"
    >
      Are you sure you want to delete this 
      Title
      ? This action cannot be undone.
    </p>
    <div
      class="delete-prompt__button-holder"
    >
      <button
        class="button white rounded-square-border border modal"
      >
        <div />
        <p
          class="text modal-text white"
        >
          Cancel
        </p>
      </button>
      <button
        class="button red rounded-square-border modal"
      >
        <div />
        <p
          class="text modal-text red"
        >
          Delete
        </p>
      </button>
    </div>
  </div>
</div>
`;
